.github/workflows/base.yml

```diff
@@ -0,0 +1,77 @@
+name: check
+on: [push, pull_request]
+jobs:
+  # The build and test job builds librdkafka, a dynamically linked version of
+  # the go client, and then runs all the tests which don't require a broker.
+  # We also build the documentation (which also tests out error generation).
+  # We can't use prebuilt librdkafka since it might not contain changes from
+  # master, and so, we have to keep the build, test, and docs steps in one job
+  # to avoid needing to build librdkafka repeatedly.
+  build-test-doc:
+    runs-on: ubuntu-latest
+    steps:
+      # Setup repositories and dependencies for confluent-kafka-go
+      - uses: actions/checkout@v3 # checks out self-repo by default
+      - uses: actions/checkout@v3
+        with:
+          repository: edenhill/librdkafka
+          path: './librdkafka'
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '=1.14'
+      - run: |
+          cd $HOME
+          go get golang.org/x/tools/cmd/godoc
+      - run: pip install beautifulsoup4
+      - run: |
+          sudo apt update
+          sudo apt install -y libcurl4-openssl-dev libssl-dev libsasl2-dev libzstd-dev
+      - run: |
+          cd librdkafka
+          ./configure --CFLAGS="-std=c99" --CXXFLAGS="-std=c++98" --install-deps --enable-devel --disable-lz4-ext --prefix=/usr/local
+      - run: |
+          cd librdkafka
+          make -j20
```

Comment:
drop 20, we don't know how many cores the build host has , just -j will make make figure it out.